File: src/main/java/spring/webmvc/domain/repository/MemberRepository.java
package spring.webmvc.domain.repository;

import java.util.Optional;

import spring.webmvc.domain.model.entity.Member;

public interface MemberRepository {

	Optional<Member> findById(Long id);

	Optional<Member> findByAccount(String account);

	boolean existsByAccount(String account);

	Member save(Member member);

	void delete(Member member);
}

File: spring-web-kotlin/src/main/kotlin/spring/web/kotlin/domain/repository/MemberRepository.kt
package spring.web.kotlin.domain.repository

import spring.web.kotlin.domain.model.entity.Member

interface MemberRepository {
    fun findByIdOrNull(id: Long): Member?
    fun findByAccount(account: String): Member?
    fun existsByAccount(account: String): Boolean
    fun save(member: Member): Member
    fun delete(member: Member)
}